mlflow/tracing/trace_manager.py

```diff
@@ -84,14 +88,23 @@ def start_detached_span(
 
         try:
             tracer = get_tracer(__name__)
-            if parent_span_id:
-                parent_span = self.get_span_from_id(request_id, parent_span_id)._span
-                context = trace_api.set_span_in_context(parent_span)
+            if parent_id:
+                parent_span = self.get_span_from_id(request_id, parent_id)
+                if parent_span is None:
+                    raise MlflowException(
+                        f"Parent span with ID '{parent_id}' not found.",
+                        error_code=RESOURCE_DOES_NOT_EXIST,
+                    )
+                context = trace_api.set_span_in_context(parent_span._span)
             else:
                 context = None
 
-            span = MlflowSpanWrapper(tracer.start_span(name, context=context), span_type=span_type)
+            otel_span = tracer.start_span(name, context=context)
 
+            if not request_id:
```

Comment:
Same comment here as before, I think: We check whether `request_id` is defined on L104, but we assume it's defined in L92. Should we move this check up / elsewhere and add a test?

Reply:
Indeed `get_span_from_id` should work even if `request_id` is `None`, but yea I think it's clear to check this first, and probably more fine-grained check e.g. if `parent_id` is passed but `request_id` is not, we shouldn't create new trace and raise. Thanks for the catch!